backend/fms_core/template_importer/row_handlers/project_link_samples/project_link_samples.py

```diff
@@ -28,18 +32,94 @@ def process_row_inner(self, sample, project, action):
                               f"is named {sample_obj.name} not {sample['sample_name']}."
                 self.warnings['sample'].append(warning_msg)
 
-            # Check if link exists to ensure there's not a duplicated association
-            if action['name'] == ADD_ACTION:
+            # Perform an add project and/or study action
+            if action['name'] == ADD_PROJECT_ACTION:
```

Comment:
This action should allow both project and study add at once. a missing study would add just the project if both are present you add both starting with project. This is a less orthogonal organization, but it will save user the extra entry that is required by separating add project and study. It also better mirror the remove project (which removes the studies).